src/plugins/intel_gpu/src/runtime/ocl/ocl_device.cpp

```diff
@@ -147,52 +150,20 @@ bool get_imad_support(const cl::Device& device) {
     return false;
 }
 
-bool is_local_block_io_supported(const cl::Device& device) {
-    try {
-        cl_int status = CL_SUCCESS;
-        cl::Context ctx(device);
-        std::string kernel_code =
-            "__attribute__((intel_reqd_sub_group_size(8)))"
-            "__attribute__((reqd_work_group_size(8, 1, 1)))"
-            "void kernel is_local_block_io_supported(global uchar* dst) {"
-            "    uint lid = get_sub_group_local_id();"
-            "    uchar val = (uchar)lid * 2;"
-            "    __local uchar tmp_slm[8];"
-            "    intel_sub_group_block_write_uc2(tmp_slm, (uchar2)(val));"
-            "    barrier(CLK_LOCAL_MEM_FENCE);"
-            "    uchar2 read = intel_sub_group_block_read_uc2(tmp_slm);"
-            "    dst[lid] = read.s0 + 1;"
-            "}";
-        cl::Program program(ctx, kernel_code);
-        if (program.build(device, "-Dcl_intel_subgroup_local_block_io") != CL_SUCCESS)
-            return false;
-        cl::Buffer buffer(ctx, CL_MEM_READ_WRITE, sizeof(uint8_t) * 8);
-        cl::Kernel kernel(program, "is_local_block_io_supported");
-        status = kernel.setArg(0, buffer);
+static std::vector<unsigned char> getProgramBinaries(cl::Program program) {
```

Comment:
Same impl is in the kernels_cache.cpp, can we move it to common header and reuse them?